SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialTraceEventDataBuilder.cpp

```diff
@@ -0,0 +1,277 @@
+const char* FSpatialTraceEventName::QueueRPCEventName = GDK_EVENT_NAMESPACE "queue_rpc";
+const char* FSpatialTraceEventName::ReceieveRPCEventName = GDK_EVENT_NAMESPACE "receive_rpc";
+const char* FSpatialTraceEventName::MergeSendRPCEventName = GDK_EVENT_NAMESPACE "merge_send_rpcs";
+const char* FSpatialTraceEventName::ApplyCrossServerRPCEventName = GDK_EVENT_NAMESPACE "apply_cross_server_rpc";
+const char* FSpatialTraceEventName::SendCommandResponseEventName = GDK_EVENT_NAMESPACE "send_command_response";
+const char* FSpatialTraceEventName::SendCommandRequestEventName = GDK_EVENT_NAMESPACE "send_command_request";
+const char* FSpatialTraceEventName::SendCrossServerRPCEventName = GDK_EVENT_NAMESPACE "send_cross_server_rpc";
+const char* FSpatialTraceEventName::ReceiveCrossServerRPCEventName = GDK_EVENT_NAMESPACE "receive_cross_server_rpc";
+const char* FSpatialTraceEventName::MergePropertyUpdateEventName = GDK_EVENT_NAMESPACE "merge_property_update";
+const char* FSpatialTraceEventName::ReceiveCreateEntityEventName = GDK_EVENT_NAMESPACE "receive_create_entity";
+const char* FSpatialTraceEventName::ReceiveRemoveEntityEventName = GDK_EVENT_NAMESPACE "receive_remove_entity";
+const char* FSpatialTraceEventName::AuthorityChangeEventName = GDK_EVENT_NAMESPACE "authority_change";
+
+// ---- FStringCache ----
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::CombineStrings(const char* A, const char* B)
+{
+	if (NextIndex >= BufferSize)
+	{
+		return BufferSize - 1;
+	}
+
+	int32 InsertIndex = NextIndex;
+	int32 InitialRemainingSize = BufferSize - NextIndex;
+	FCStringAnsi::Strncpy(&Buffer[NextIndex], A, InitialRemainingSize);
+
+	int32 CharSize = sizeof(char);
+	int32 ALength = FCStringAnsi::Strlen(A);
+	int32 RemainingSize = InitialRemainingSize - ALength;
+	RemainingSize = FMath::Max(0, RemainingSize);
+
+	if (RemainingSize > CharSize)
+	{
+		FCStringAnsi::Strncpy(&Buffer[NextIndex + ALength], B, RemainingSize);
+	}
+
+	int32 BLength = FCStringAnsi::Strlen(B);
+	RemainingSize = RemainingSize - BLength - CharSize;
+	RemainingSize = FMath::Max(0, RemainingSize);
+
+	NextIndex += InitialRemainingSize - RemainingSize;
+	return InsertIndex;
+}
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::AddString(const char* String)
+{
+	if (NextIndex >= BufferSize)
+	{
+		return BufferSize - 1;
+	}
+
+	int32 InsertIndex = NextIndex;
+	int32 InitialRemainingSize = BufferSize - NextIndex;
+	FCStringAnsi::Strncpy(&Buffer[NextIndex], String, InitialRemainingSize);
+
+	int32 CharSize = sizeof(char);
+	int32 StringLength = FCStringAnsi::Strlen(String);
+	int32 RemainingSize = InitialRemainingSize - StringLength - CharSize;
+	RemainingSize = FMath::Max(0, RemainingSize);
+
+	NextIndex += InitialRemainingSize - RemainingSize;
+	return InsertIndex;
+}
+
+int32 FSpatialTraceEventDataBuilder::FStringCache::AddFString(const FString& String)
+{
+	std::string ValueSrc = (const char*)TCHAR_TO_ANSI(*String);
+	return AddString(ValueSrc.c_str());
```

Comment:
I don't think `std::string` is required here
```suggestion
	return AddString( (const char*)TCHAR_TO_ANSI(*String)));
```